### ISSUE TITLE
[question] how to proceed with the amount of patches growing -> push them upstream

### ISSUE THREAD

**Issue by gocarlos** [conan-io/conan-center-index#2859]
Currently a lot of packages need patches for the package to be usable on the different platforms and with the different options.

Most of those patches (no my knowledge) are not upstreamed, this means that they have to be maintained for ever.
With newer releases, this becomes a big effort...

Having patches downstream is not something new, debian and yocto do the same as conan, though one difference seems to be that at least with yocto patches, there is a note at the beginning of the patch with a link to the MR/mailing list where the patch was posted in order to push it upstream.

**Request**: add upstream status to newer patches and ask people to upstream patches.

**Comment by Croydon** [comment]
I agree with this. A few of us, me included, did ask for upstreaming for a while now, since I guess we all had the unpleasant experience of rebasing patches for newer versions...

If we can can formalize this somehow and add the information directly to the recipe files, then this would be even better than just having those information in some pull request which is hard to find again

**Comment by gocarlos** [comment]
@Croydon see this link for clarification (this is what I think CCI patches need): https://www.yoctoproject.org/pipermail/yocto/2011-April/001237.html

**Comment by uilianries** [comment]
We encourage pushing to the upstream of course, avoiding separated patches in CCI would be great.
